src/NUnitFramework/framework/Constraints/Operators/ExactCountOperator.cs

```diff
@@ -29,7 +29,7 @@ namespace NUnit.Framework.Constraints
     /// Represents a constraint that succeeds if the specified 
     /// count of members of a collection match a base constraint.
     /// </summary>
-    public class ExactCountOperator : CollectionOperator
+    public class ExactCountOperator : SelfResolvingOperator
```

Comment:
Ideally, this should be a CollectionOperator, which would eliminate some of the hacks I had to add. However, it also needs to be a SelfResolvingOperator but it can't be both. In the future, we should probably eliminate SelfResolvingOperator since it's only used as a marker. Operators that are self-resolving could indicate it by use of a  property.